### PR TITLE
added ajax-hw

### PR DESCRIPTION
**Pull request by j-tws**
_If you had no problems you can just leave it empty_

- How difficult did you find this (out of 10)? (0 being no problems at all, 10 being impossible)
ajax hw - 7.5
axios hw - 5.5

- Was there anything that you struggled with?
**AJAX HW** - I don't know how to dynamically show movie details and switch to movie search results whilst remain in the same page. Totally no idea how :(
**AXIOS HW** - A bit more manageable after Luke did the homework solution for Ajax hw

- Is there any particular code you want me to look into?
yes i want to know solutions (or at least how to)

- Is there anything that you'd like some further information on?
- Roughly how long did it take?
3-4hours for both ajax-hw and axios hw

### CODE REVIEW

File: justin-tan/week7/wed/flickr-search/js/main.js [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+            getNextPageForScrolling(pageNum += 1, query)
+        }
+
+        //Take the difference of document height and window height. The difference is equal to window scrollTop when the user has scrolled to the bottom.
+        // $(document).height : It's the height of the html document
+        // window.height : It's the height of browser viewport
+        // window.scrollTop : returns the vertical scrollbar position for the element.
+    
+    })
+
```
Comment (kristabel-wong) [inline justin-tan/week7/wed/flickr-search/js/main.js:87]: Great work on infinite scroll!

File: justin-tan/week7/wed/flickr-search/js/main.js [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+                <a href="#" id="${i}" class="pageTag">${i}</a>
+            `)
+        }
+    }
+    
+    $('.pageTag').on('click', function(){
+        const pageId = $('.pageTag').attr('id')
+        console.log(`page button clicked!`, pageId, query);
+        // getPage(pageId, query)
+    })
```
Comment (kristabel-wong) [inline justin-tan/week7/wed/flickr-search/js/main.js:64]: doesn't work :'(?

File: justin-tan/week7/wed/flickr-search/js/main.js [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+
+    // EITHER a click of the button OR (yessss) pressing enter in an input
+    // will trigger the form submit
+    $('#searchForm').on('submit', function( ev ){
+        ev.preventDefault(); // stop the form from reloading the page
+
+        query = $('#query').val();
+
+        getSearchResults( query )
+
```
Comment (kristabel-wong) [inline justin-tan/week7/wed/flickr-search/js/main.js:21]: you should add a `show()` and `hide()` function here for the previous page and next page so they only show when a query has been entered

**Review by kristabel-wong — COMMENTED** [review]
overall good work! love the api you used and what you did with it!

**Comment by j-tws** [comment]
Why are you up marking hw at a weekend night? Go party or sleep!

Tho I do appreciate it thanks heaps!

File: justin-tan/week7/wed/flickr-search/js/main.js (hunk 2 above)
Comment (j-tws) [reply justin-tan/week7/wed/flickr-search/js/main.js:64]: Got it working today! Just need to push it

File: justin-tan/week7/wed/flickr-search/js/main.js (hunk 3 above)
Comment (j-tws) [reply justin-tan/week7/wed/flickr-search/js/main.js:21]: True that. I'll give that a go

**Pull request merged**